apple/testing/default_runner/ios_test_runner.bzl

```diff
@@ -28,6 +28,11 @@ def _get_template_substitutions(ctx):
         "testrunner_binary": ctx.executable._testrunner.short_path,
         "device_type": ctx.attr.device_type,
     }
+    # if ios_simulator_device and ios_simulator_version are specified in the bazel test command, override device_type
```

Comment:
overriding the `device_type` like this is problematic, as debugging will get tricky when a test is running on an iPhone 6 when the `device_type` of the test runner is iPhone 7. another problem is overriding the device type for all tests in  running tests with `//...`, as it will override any device configured in the targets that you might have in your workspace

the approach I think this should take is to resolve `device_type` like this:

1. if `device_type` attribute is set, use that.
2. if `device_type` is not set, use the `ios_simulator_device` from the `objc` fragment
3 if the `ios_simulator_device` is not set, use a default of "iPhone 6" (for example)

same approach with `os_version`:

1. if `os_version` is set, use that
2. if `os_version` not set, use the `ios_simulator_version` value from the `objc` fragment
3. if `ios_simulator_version` is not set, use the iOS SDK value from the `apple_common.XcodeVersionConfig` provider

Reply:
Great idea. I will update the PR